crates/typed-store-macros/src/lib.rs

```diff
@@ -409,3 +410,217 @@ fn extract_generics_names(generics: &Generics) -> Vec<Ident> {
+        .iter()
+        .map(|q| {
+            let GeneralTableOptions::OverrideFunction(fn_name) = q;
+            fn_name.parse().unwrap()
+        })
+        .collect();
+
+    let generics_bounds =
+        "core::hash::Hash + core::cmp::Eq + std::fmt::Debug + serde::Serialize + for<'de> serde::de::Deserialize<'de> + core::marker::Send + 'static";
+    let generics_bounds_token: proc_macro2::TokenStream = generics_bounds.parse().unwrap();
+
+    let config_struct_name_str = format!("{}Configurator", name);
+    let config_struct_name: proc_macro2::TokenStream = config_struct_name_str.parse().unwrap();
+
+    // let first_field_name = field_names
+    //     .get(0)
+    //     .expect("Expected at least one field")
+    //     .clone();
+
+    let _simple_field_type_name: proc_macro2::TokenStream = simple_field_type_name.parse().unwrap();
+
+    TokenStream::from(quote! {
+        /// Create config structs for configuring Store tables
+        pub struct #config_struct_name {
+            #(
+                pub #field_names : rocksdb::Options,
+            )*
+        }
+
+        impl #config_struct_name {
+            /// Initialize to defaults
+            pub fn init() -> Self {
+                Self {
+                    #(
+                        #field_names : typed_store::rocks::default_rocksdb_options(),
+                    )*
+                }
+            }
+
+            /// Build a config
+            pub fn build(&self) -> typed_store::rocks::DBMapTableConfigMap {
+                typed_store::rocks::DBMapTableConfigMap::new([
+                    #(
+                        (stringify!(#field_names).to_owned(), self.#field_names.clone()),
+                    )*
+                ].into_iter().collect())
+            }
+        }
+
+        impl <
+                #(
+                    #generics_names: #generics_bounds_token,
+                )*
+            > #name #generics {
+
+                pub fn configurator() -> #config_struct_name {
+                    #config_struct_name::init()
+                }
+        }
+        impl <
+                #(
+                    #generics_names: #generics_bounds_token,
+                )*
+            > StoreTableUtil for #name #generics{
+            /// Opens a set of tables in read-write mode
+            /// Only one process is allowed to do this at a time
+            /// `global_db_options_override` apply to the whole DB
+            /// `tables_db_options_override` apply to each table. If `None`, the attributes from `default_options_override_fn` are used if any
+            fn open_tables_read_write(
+                path: std::path::PathBuf,
+                global_db_options_override: Option<rocksdb::Options>,
+                tables_db_options_override: Option<typed_store::rocks::DBMapTableConfigMap>
+            ) -> Self {
+                Self::open_tables_impl(path, None, global_db_options_override, tables_db_options_override)
+            }
```

Comment:
This strongly hints at the idea that `StoreTableUtil` could be a trait with one required method (`open _tables_impl`), and a few methods with default implementations, such as this one. Then, the macro would only have to generate the required method in the impl.